Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,344 @@
+            {
+                if ( interactables.Count > 0 )
+                {
+                    GameObject[] interactableList = interactables.ToArray();
+
+                    for ( int i = 0; i < interactableList.Length; i++ )
+                    {
+                        if ( interactableList[ i ] != null )
+                        {
+                            Gizmos.color = Color.green;
+                            Gizmos.DrawLine( transform.position, interactableList[ i ].transform.position );
+                        }
+                    }
+                }
+
+                if ( Targets.Count > 0 )
+                {
+                    GameObject[] targetList = Targets.ToArray();
+
+                    for ( int i = 0; i < targetList.Length; i++ )
+                    {
+                        if ( targetList[ i ] != null )
+                        {
+                            Gizmos.color = Color.red;
+                            Gizmos.DrawLine( transform.position, targetList[ i ].transform.position );
+                        }
+                    }
+                }
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Register an interactable with this receiver.
+        /// </summary>
+        /// <param name="interactable">takes a GameObject as the interactable to register.</param>
+        public virtual void RegisterInteractable( GameObject interactable )
+        {
+            if ( interactable == null || interactables.Contains( interactable ) )
+            {
+                return;
+            }
+
+            interactables.Add( interactable );
+        }
+
+        /// <summary>
+        /// Function to remove an interactable from the linked list.
+        /// </summary>
+        /// <param name="interactable"></param>
+        public virtual void RemoveInteractable( GameObject interactable )
+        {
+            if ( interactable != null && interactables.Contains( interactable ) )
+            {
+                interactables.Remove( interactable );
+            }
+        }
+
+        /// <summary>
+        /// Clear the interactables list and unregister them
+        /// </summary>
+        public virtual void ClearInteractables()
+        {
+            GameObject[] _intList = interactables.ToArray();
+
+            for ( int i = 0; i < _intList.Length; i++ )
+            {
+                RemoveInteractable( _intList[ i ] );
+            }
+        }
+
+        /// <summary>
+        /// Is the game object interactable in our list of interactables
+        /// </summary>
+        /// <param name="interactable"></param>
+        /// <returns></returns>
+        protected bool IsInteractable( GameObject interactable )
+        {
+            return ( interactables != null && interactables.Contains( interactable ) );
+        }
+
+        #region Global Listener Callbacks
+
+        public void OnBeforeFocusChange( FocusEventData eventData ) { /*Unused*/ }
```

Comment:
I know this is a bit nit picky but let's be sure to keep the spacing consistent as well in our project. 

Please, also add the `/// <Ineritdoc/>` xml tag for any interface implementations.